hack/development-environment/README.md

```diff
@@ -0,0 +1,34 @@
+# A canonical Development Environment for Calico
+
+This directory takes as input every calico source dependency (See DEVELOPER_GUIDE.md for details on those), and assumes they
+are one level above the calico repository.
+
+It then uses vagrant to 
+
+- create a linux vm
+- build all calico images from calico/ ecosystem projects
+- mounts these into /calico/all
+- runs the `make` targets for creating all docker images for calico
+- makes calico orchestration files (i.e. calico.yaml)
+- installs a quick kubeadm (single node) and sets relevant iptables/swap rules
+- installs the new images that were made from your current branch
+- smoke tests that all calico containers are running.
+
+# How to use this recipe
+
+- You can use the `vagrant up` command to test that the changes you made to any calico repository wont break CI,
+or reproduce a failure in CI.
+- You can use `vagrant up` followed by `vagrant ssh` to get into a kube cluster running the exact source of your
+calico build.
+- You can use the install.sh script as a quick start for your own dev worklow or automation tooling if you are using
+your own calico builds in house. 
+
+# What this recipe is not
+
+- This is not a CI system for upstream calico, for that, see the contributor and developer docs.
+- This is not a full test suite: It doesnt run any kind of unit or performance tests.  Patches
+are welcome to extendt the install.sh with other optional test/make targets, but ideally, the smoke
```

Comment:
s/extendt/extend

p.s. was trying this out locally, couldn't get it to work for me, maybe I'm doing something wrong 😅 

<img width="725" alt="image" src="https://user-images.githubusercontent.com/5788514/67049122-7fa18b00-f103-11e9-84af-0b543604fd9a.png">

I did put all the required repos in the path:

<img width="707" alt="image" src="https://user-images.githubusercontent.com/5788514/67049181-a19b0d80-f103-11e9-96a0-f1ea3bf02153.png">

Reply:
haha look its yash ! hey buddy can you send me the *FULL OUTPUT* ? it looks to me like literally none of the images built .

Reply:
Actually got further, and got the images to build successfully.  My machine just doesn't seem to have enough juice for the VM it spins up.  Probably, can make a slimer version of this in a follow-up PR 😄 

This is pretty cool tho 🤙 

Concept ACK!

<img width="1680" alt="image" src="https://user-images.githubusercontent.com/5788514/67051801-2c7f0680-f10a-11e9-870f-e6c7692a09a9.png">

Reply:
Gotcha, marking resolved.

Yash feel free to try a build w 2 core/8g and if it works we can lower the defaults.

Reply:
If it makes you feel better, it took a while on my machine, but I did finally get it to work 😄 

<img width="1680" alt="image" src="https://user-images.githubusercontent.com/5788514/67060549-57c61d80-f12b-11e9-9225-ae70e0f774b4.png">

```
diff --git a/hack/development-environment/Vagrantfile b/hack/development-environment/Vagrantfile
index 22bc3939..f6ddd55b 100644
--- a/hack/development-environment/Vagrantfile
+++ b/hack/development-environment/Vagrantfile
@@ -14,8 +14,8 @@ Vagrant.configure("2") do |config|
       ip = "172.17.8.#{i+100}"
       node.vm.network "private_network", ip: ip
       node.vm.provider "virtualbox" do |vb|
-        vb.memory = "16000"
-        vb.cpus = 3
+        vb.memory = "4000"
+        vb.cpus = 2
         vb.name = "node#{i}"
       end
       if (node.vm.hostname == "node1")
```

Reply:
ok, lets split the difference @ 2 and 8 :) thanks for testing this !